core/src/main/scala/pureconfig/DerivedWriters.scala

```diff
@@ -12,6 +12,7 @@ trait DerivedWriters extends DerivedWriters1 {
   implicit def deriveAnyVal[T, U](
     implicit
     ev: T <:< AnyVal,
+    generic: Generic[T],
```

Comment:
The important part is that we don't derive `ConfigReader`s and `ConfigWriter`s for classes that have no generic instance because `Unwrapped` there is defined as the identity.